### DISCUSSION TITLE
Future of v4?

### DISCUSSION THREAD

**Discussion by zehawki** [react-page/react-page#1240] [General] [answered]
@macrozone would you pls comment on the future of v4 branch vs v5. For the near future I don't expect to move to MUI5, and hence would like to stay with the corresponding v4 for react-page. Is it a lot of trouble for you to release important updates on v4? Maybe this is important to others as well.

As a special note - since my environment is now finally up (yeah!), I would like to submit PRs. Should I do this on v4 or v5? If its v5, will it make its way into v4? (I'm not very good at the PR process, so I'm not even sure if one can submit PRs on the older v4.)

**Comment by macrozone** [comment, marked as answer]
@zehawki v4 won't receive any more updates, but you can still use it alongside mui4 if the rest of your app uses mui4. Its documented somewhere in the documentation.

New MRs should just target the default branch (currently master). Merges on master trigger a new release and bump the version (depending on the type of commits). updates on older tags (v4) is possible, but can be tedious to do and should be limited to serious bugfixes.

**Reply by zehawki** [reply]
>updates on older tags (v4) is possible, but can be tedious to do and should be limited to serious bugfixes

As an example, I'll submit a PR for this https://github.com/react-page/react-page/issues/987. Would you consider this a "serious bugfix"?

**Reply by macrozone** [reply]
I first suggest to just use the latest version as it is compatible with react 17, react 18, mui4 and mui5. The goal is anyway to have mui only as an implementation detail, its a bit unfortunate that it sometimes conflicts with itself.

the semantic-release we are using supports doing updates for legacy versions, but i think it needs a bit tweaking as far as i remember.  The goal here is that if someone really needs to have an old version patched, they can provide a Pull request (branched from that tag) and we try to get it released.
